scripts/ci/docker_upgrade.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+set -e
+
+check_parameters() {
+  echo "-- check_parameters"
+
+  if [ -z "${IMAGE_NAME}" ] || [ -z "${IMAGE_VARIANT}" ] || [ -z "${DOCKER_REPOSITORY}" ]; then
+    echo "\${IMAGE_NAME}, \${IMAGE_VARIANT} and \${DOCKER_REPOSITORY} need to be defined."
+
+    false
+  fi
+}
+
+check_version() {
+  echo "-- check_version"
+
+  git fetch
+  git branch
```

Comment:
```suggestion
  git fetch origin
  git branch origin
```

Reply:
Actually, do we need these at all?

Reply:
We do but I think it actually belongs in the .yml file, together with the checkout. I would not be looking for it here.

The checkout action does a shallow clone without remote branches by default. You only have the one that is checked out. So if we want to keep using the checkout action we either need this or a deep clone.

But I'd just replace the checkout action with
``` bash
git clone --depth 1 --no-single-branch --branch "${{ github.head_ref }}" "https://github.com/${GITHUB_REPOSITORY}.git"
```
to still have a shallow clone but with tips of all the remote branches instead.

---

`git branch` is here only to print the name of all the local branches I think. BTW, I just noticed that I added `origin` to it too in my suggestion. I updated the suggestion.

Reply:
Ah, ok, I see. You're definitely right then, it rather belongs in the action. None of these here need to be optimized for speed though, so configuring the checkout action with `fetch-depth: 0` would also be fine and maybe simpler as a manual checkout - both seems fine to me though.